### ISSUE TITLE
[Feedback Wanted] User permissions

### ISSUE THREAD

**Issue by vinaykapadia** [bhj/react-redux-starter-kit#58]
Not sure what to call it, but add an account type thats between User and Admin. Someone that can create rooms, start player, and edit the queue for the rooms they create, but can't create or edit users, or change the media folders.

Basically on the account page, out of Rooms, Users, Preferences, and My Account, User can access only My Account, Admin can access all, and this new account type can access My Account and Rooms.

This is useful because I can then have someone set up a room and run it without having to give them access to full admin features.

**Comment by bhj** [comment]
Definitely want this! We've gotten a lot of mileage out of the `isAdmin` flag, but it's probably time to move to a system under the hood that would support arbitrary permissions and granularity (even if the UI on top stays simple).

Let's use this thread to hear wants/needs around user permissions. Thanks!

**Comment by vinaykapadia** [comment]
I wanted to start working on some concepts, but it took me, no joke, like 4 hours to get the code to even run on my machine (my node.js was all kinds of messed up and I'm not too familiar with it either). Finally got it running though. Will try some things tomorrow.

**Comment by vinaykapadia** [comment]
I don't think we need more granularity than three account types, but maybe if we want it in the future we can prep now by setting up permissions and roles. But I also don't want to overdo it if it's not necessary.

**Comment by MercifulCamel** [comment]
Would also be nice if it was possible to set a default account type on newly created accounts.
It would help greatly if I could set the default account type to one that will be able to create rooms, start songs etc

**Comment by threeeye** [comment]
I know this is a 2.5 year old suggestion, but is there any plan on implementing this?

**Comment by bhj** [comment]
> I know this is a 2.5 year old suggestion, but is there any plan on implementing this?

In v2 there is a roles table, so the foundation now exists! Not sure how granular they should ultimately be, but I'm in favor of this.

**Comment by thomasesr** [comment]
https://github.com/bhj/KaraokeEternal/pull/90 adds Room Manager role.

**Comment by vinaykapadia** [comment]
Awesome, thanks!

**Comment by threeeye** [comment]
Thank you!
I've been waiting for this!
Can't wait to try it out
